sdks/python/apache_beam/runners/worker/statesampler_slow.py

```diff
@@ -49,19 +55,15 @@ def start(self):
   def stop(self):
     self.finished = True
 
-  def get_info(self):
-    """Returns StateSamplerInfo with transition statistics."""
-    return StateSamplerInfo(
-        self.current_state().name, self.transition_count, 0)
-
 
 class ScopedState(object):
 
-  def __init__(self, sampler, name, counter=None):
+  def __init__(self, sampler, name, counter=None, metrics_container=None):
     self.state_sampler = sampler
     self.name = name
     self.counter = counter
     self.nsecs = 0
+    self.metrics_container = metrics_container
```

Comment:
Adding a counter container here seems out-of-place, and who knows what else might get added.

Instead, I'd suggest an architecture where the there is the ability to get the local state (which is an object that can be used as a key and has basic information like step name) which other frameworks (e.g. metrics) consume in a self-contained way. This also reduces the scope of the global state we must store to access these contexts implicitly from user code.

Reply:
I agree that tagging state here may be useful for cleanup when the state is no longer needed, so I suppose this is OK.